pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -13,25 +15,69 @@ import (
 	"vitess.io/vitess/go/vt/wrangler"
 )
 
+const (
+	maxSafeVReplicationLag = 10
+)
+
 func (r *ReconcileVitessKeyspace) reconcileResharding(ctx context.Context, vtk *planetscalev2.VitessKeyspace) error {
 	wr, ts, err := newWrangler(ctx, vtk.Spec.GlobalLockserver)
 	if err != nil {
 		return err
 	}
 	defer ts.Close()
 
-	workflows, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
+	workflowList, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
 	if err != nil {
-		workflows = make([]string, 0)
+		workflowList = make([]string, 0)
 	}
 
-	vtk.Status.ActiveWorkflows = workflows
-	// TODO: Address when ConditionUnknown is applicable.
-	if len(workflows) != 0 {
-		vtk.Status.ReshardingInProgress = corev1.ConditionTrue
-	} else {
-		vtk.Status.ReshardingInProgress = corev1.ConditionFalse
+	reshardingInProgress := corev1.ConditionUnknown
+	workflows := make([]planetscalev2.WorkflowStatus, 0, len(workflowList))
+	for _, workflowName := range workflowList {
```

Comment:
We should skip over workflows that are not resharding workflows (source/target keyspace is the same, but source/target shard sets are not exactly the same).

Reply:
Done.